src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -3614,57 +3614,584 @@ struct S
+class Derived : Base
+{
+    private string _field;
+    private Derived()
+    {
+    }
+}
+""";
+
+        var expectedDiagnostics = new[] {
+            // (4,20): warning CS0169: The field 'Derived._field' is never used
+            //     private string _field;
+            Diagnostic(ErrorCode.WRN_UnreferencedField, "_field").WithArguments("Derived._field").WithLocation(4, 20),
+            // (5,13): warning CS8618: Non-nullable field '_field' must contain a non-null value when exiting constructor. Consider declaring the field as nullable.
+            //     private Derived()
+            Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "Derived").WithArguments("field", "_field").WithLocation(5, 13)
+        };
+
+        var comp = CreateCompilationWithRequiredMembers(new[] { @base, derived, MemberNotNullAttributeDefinition });
+        comp.VerifyDiagnostics(expectedDiagnostics);
+
+        var baseComp = CreateCompilationWithRequiredMembers(new[] { @base, MemberNotNullAttributeDefinition });
+        comp = CreateCompilation(derived, new[] { baseComp.EmitToImageReference() });
+        comp.VerifyDiagnostics(expectedDiagnostics);
+    }
+
+    [Fact, CompilerTrait(CompilerFeature.NullableReferenceTypes)]
+    [WorkItem(6754, "https://github.com/dotnet/csharplang/issues/6754")]
+    public void RequiredMemberSuppressesNullabilityWarnings_MemberNotNull_ChainedBaseConstructor_05()
+    {
+        var @base = """
+using System.Diagnostics.CodeAnalysis;
+#nullable enable
+public class Base 
+{
+    protected string _field;
+    [MemberNotNull(nameof(_field))] public required string Property { get => _field ??= ""; set => _field = value; }
+
+    public Base() { }
+}
+""";
+
+        var derived = """
+using System.Diagnostics.CodeAnalysis;
+#nullable enable
+class Derived : Base
+{
+    [SetsRequiredMembers]
+    public Derived()
+    {
+        _field.ToString();
+    }
+}
+""";
+
+        var expectedDiagnostics = new[] {
+            // (6,12): warning CS8618: Non-nullable property 'Property' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
+            //     public Derived() { }
+            Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "Derived").WithArguments("property", "Property").WithLocation(6, 12),
+            // (8,9): warning CS8602: Dereference of a possibly null reference.
+            //         _field.ToString();
+            Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "_field").WithLocation(8, 9)
+        };
+
+        var comp = CreateCompilationWithRequiredMembers(new[] { @base, derived, MemberNotNullAttributeDefinition });
+        comp.VerifyDiagnostics(expectedDiagnostics);
+
+        var baseComp = CreateCompilationWithRequiredMembers(new[] { @base, MemberNotNullAttributeDefinition });
+        comp = CreateCompilation(derived, new[] { baseComp.EmitToImageReference() });
+        comp.VerifyDiagnostics(expectedDiagnostics);
+    }
```

Comment:
Consider adding a test which demonstrates that a transitive closure is not performed, e.g. in the following, the constructor analysis doesn't trace the idea that setting Prop1 sets Prop2 sets Prop3.

```cs
public class C
{
    public required string Prop1 { get => Prop2; [MemberNotNull(nameof(Prop2))] set => Prop2 = value; }
    public string Prop2 { get => Prop3; [MemberNotNull(nameof(Prop3))] set => Prop3 = value; }
    public string Prop3 { get; set; }
    public C() // warning CS8618: Non-nullable property 'Prop3' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.

    {
    }
}
```